gear-test/spec/test_create_program.yaml

```
@@ -0,0 +1,128 @@
+title: Test create_program sys-call
+
+codes:
+  # Save child's code
+  - path: examples/create-program/child_contract.wasm
+
+programs:
+  - id: 1
+    source:
+      kind: id
+      value: 101
+    path: target/wasm32-unknown-unknown/release/demo_create_program.wasm
+
+fixtures:
+  - title: Simple creation of program from program
+
+    messages:
+      - destination: 1
+        source:
+          kind: id
+          value: 100
+        payload:
+          kind: utf-8
+          value: default
+
+    expected:
+      - programs:
+          - kind: id
+            value: 1
+
+          - kind: ss58
+            # computed manually new program address
+            value: 5Fypuf8SCGzqtJy91VJqb17WgUbgVTC6k3Vb4yfj3VBj2at1
+
+  - title: Try to re-init existing program
+
+    messages:
+      - destination: 1
+        source:
+          kind: id
+          value: 100
+        payload:
+          kind: utf-8
+          value: default
+
+      - destination: 1
+        source:
+          kind: id
+          value: 100
+        payload:
+          kind: utf-8
+          value: duplicate    
+
+    expected:
+      - step: 1
+        messages:
+          - destination: 1
+            source:
+              kind: id
+              value: 100
+            payload:
+              kind: utf-8
+              value: duplicate
+          - destination: 
+              kind: ss58
+              value: 5Fypuf8SCGzqtJy91VJqb17WgUbgVTC6k3Vb4yfj3VBj2at1
+            init: true
+            payload:
+              kind: utf-8
+              value: unique
+          - destination: 
+              kind: ss58
+              value: 5Fypuf8SCGzqtJy91VJqb17WgUbgVTC6k3Vb4yfj3VBj2at1
```

We have anchors in YAML for such kind repeating objects, please take it into account in the future.
```suggestion
          - destination: &address
              kind: ss58
              value: 5Fypuf8SCGzqtJy91VJqb17WgUbgVTC6k3Vb4yfj3VBj2at1
```

---

done